src/passes/OnceReduction.cpp

```diff
@@ -339,6 +356,14 @@ struct OnceReduction : public Pass {
       // Fill in the map so that it can be operated on in parallel.
       optInfo.onceFuncs[func->name] = Name();
     }
+    for (auto& ex : module->exports) {
+      if (ex->kind == ExternalKind::Global) {
+        // An exported global cannot be "once" since the outside may read and
+        // write to it in ways we are unaware.
+        // TODO: This could perhaps only be on mutable ones.
```

Comment:
Why don't we check that now, given that it should be simple..?

Reply:
It is simple to check, but really only mutable globals are relevant here - a mutable one would never fit the pattern we look for. I think such globals would be optimized by other passes already. I'll improve the comment here.